x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -53,21 +53,21 @@ public CrossClusterAccessAuthenticationService(
     }
 
     public void authenticate(final String action, final TransportRequest request, final ActionListener<Authentication> listener) {
-        final Authenticator.Context authcContext = authenticationService.newContext(action, request, false);
-        final ThreadContext threadContext = authcContext.getThreadContext();
-
+        final ThreadContext threadContext = clusterService.threadPool().getThreadContext();
         final CrossClusterAccessHeaders crossClusterAccessHeaders;
+        final ApiKeyService.ApiKeyCredentials apiKeyCredentials;
         try {
             // parse and add as authentication token as early as possible so that failure events in audit log include API key ID
             crossClusterAccessHeaders = CrossClusterAccessHeaders.readFromContext(threadContext);
-            final ApiKeyService.ApiKeyCredentials apiKeyCredentials = crossClusterAccessHeaders.credentials();
+            apiKeyCredentials = crossClusterAccessHeaders.credentials();
             assert ApiKey.Type.CROSS_CLUSTER == apiKeyCredentials.getExpectedType();
-            authcContext.addAuthenticationToken(apiKeyCredentials);
```

Comment:
All the changes here (to `CrossClusterAccessAuthenticationService`), are implemented in order to avoid this `authcContext.addAuthenticationToken(apiKeyCredentials)` call. Instead the `apiKeyCredentials` token is passed in to the context constructor.